monkey/monkey_island/cc/services/attack/technique_reports/T1136.py

```diff
@@ -12,27 +12,26 @@ class T1136(AttackTechnique):
     scanned_msg = "Monkey tried creating a new user on the network's systems, but failed."
     used_msg = "Monkey created a new user on the network's systems."
 
+    query = [{'$match': {'telem_category': 'post_breach',
+                         '$or': [{'data.name': POST_BREACH_BACKDOOR_USER},
+                                 {'data.name': POST_BREACH_COMMUNICATE_AS_NEW_USER}]}},
+             {'$project': {'_id': 0,
+                           'machine': {'hostname': '$data.hostname',
+                                       'ips': ['$data.ip']},
+                           'result': '$data.result'}}]
+
     @staticmethod
     def get_report_data():
         data = {'title': T1136.technique_title()}
 
-        scanned_nodes = ReportService.get_scanned()
-        status = ScanStatus.UNSCANNED.value
-        for node in scanned_nodes:
-            if node['pba_results'] != 'None':
-                for pba in node['pba_results']:
-                    if pba['name'] in [POST_BREACH_BACKDOOR_USER,
-                                       POST_BREACH_COMMUNICATE_AS_NEW_USER]:
-                        status = ScanStatus.USED.value if pba['result'][1]\
-                                                       else ScanStatus.SCANNED.value
-                        data.update({
-                            'info': [{
-                                'machine': {
-                                    'hostname': pba['hostname'],
-                                    'ips': node['ip_addresses'],
-                                },
-                                'result': ': '.join([pba['name'], pba['result'][0]])
-                            }]
-                        })
-            data.update(T1136.get_base_data_by_status(status))
+        create_user_info = list(mongo.db.telemetry.aggregate(T1136.query))
+
+        status = []
+        for pba_node in create_user_info:
+            status.append(pba_node['result'][1])
```

Comment:
It would be better if we could do this check through mongodb query. Also, notice that most other PR's have the exact same code as you have here. A sign you should refactor at some point.

Reply:
How exactly would we check this through a mongodb query?

From what I understand, we can use queries to fetch data according to certain conditions (PBA name, here). But after that, we have to match certain conditions over all the entries _within_ the fetched data (`create_user_info`) to decide what value to assign to another variable (`status`). Is it possible to do that using a query?

Reply:
By doing a query that returns the count of PBA's successful and failed results. This would be more readable, since you could name what you're returning (SuccessfulPbaAttemps instead of pba_node['result'][1] and wouldn't need this code). Another way would be to create a function that would sound something like "get_status_from_cmd_results", put it into `AttackTechnique` and reuse throughout most of our PBA's. 
You can also do both or neither, no big deal, your code is already beyond being "good enough". The important thing in CR is to understand and exercise our sense of clean code and design choices.

Reply:
@VakarisZ just pushed some changes. Did you mean something like this?